diff --git a/3rdparty/stout/include/stout/os/open.hpp b/3rdparty/stout/include/stout/os/open.hpp
index ceff89f..d9bb17c 100644
--- a/3rdparty/stout/include/stout/os/open.hpp
+++ b/3rdparty/stout/include/stout/os/open.hpp
@@ -38,8 +38,7 @@
 // Copied from '/usr/include/sys/fcntl.h'
 #define O_CLOEXEC 0x1000000
 #elif defined(__linux__)
-// Copied from '/usr/include/asm-generic/fcntl.h'.
-#define O_CLOEXEC 02000000
+#include <fcntl.h>
 #elif defined(__sun)
 // Not defined on Solaris, taking a spare flag.
 #define O_CLOEXEC 0x1000000
