diff --git a/Makefile b/Makefile
index 603692c..2908756 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ WITH_SECCOMP ?= yes
 export prefix      = /usr
 export exec_prefix = $(prefix)
 export bindir      = $(exec_prefix)/bin
-export libdir      = $(exec_prefix)/lib
+export libdir      = $(exec_prefix)/lib64
 export docdir      = $(prefix)/share/doc
 export libdbgdir   = $(prefix)/lib/debug$(libdir)
 export includedir  = $(prefix)/include
