--- spral_nvcc_lib.m4	2017-05-28 11:40:15.110846674 +0200
+++ spral_nvcc_lib_manual.m4	2017-06-02 12:20:02.632762045 +0200
@@ -14,7 +14,8 @@
 AC_REQUIRE([SPRAL_PROG_NVCC])
 
 # Check in default path first
-NVCC_INCLUDE_FLAGS="$NVCC_INCLUDE_FLAGS"
+NVCC_INCLUDE_FLAGS="/opt/cuda/include" 
+#$NVCC_INCLUDE_FLAGS"
 AC_CHECK_HEADER([cuda_runtime_api.h], [spral_nvcc_inc_ok=yes])
 
 # Check in CUDA_HOME/include
