diff --git a/configure b/configure
index d121c6f..72bd576 100755
--- a/configure
+++ b/configure
@@ -52,11 +52,12 @@ while true; do
     # Going through one more level of expansion to handle that.
     CUDNN_INSTALL_PATH=$(bash -c "readlink -f $CUDNN_INSTALL_PATH")
   fi
-  if [ -e "$CUDNN_INSTALL_PATH/libcudnn.so.6.5" -o -e "$CUDNN_INSTALL_PATH/lib64/libcudnn.so.6.5" ]; then
+  if [ -e "$CUDNN_INSTALL_PATH/libcudnn.so.6.5" -o -e "$CUDNN_INSTALL_PATH/lib64/libcudnn.so.6.5" -o -e "${CUDNN_INSTALL_PATH}/lib/libcudnn.so.6.5" ]; then
     break
   fi
-  echo "Invalid path to CUDNN 6.5 V2 toolkit. Neither of the following two files can be found:"
+  echo "Invalid path to CUDNN 6.5 V2 toolkit. Neither of the following files can be found:"
   echo "$CUDNN_INSTALL_PATH/lib64/libcudnn.so.6.5"
+  echo "$CUDNN_INSTALL_PATH/lib/libcudnn.so.6.5"
   echo "$CUDNN_INSTALL_PATH/libcudnn.so.6.5"
   if [ -z "$fromuser" ]; then
     exit 1

diff --git a/third_party/gpus/cuda/cuda_config.sh b/third_party/gpus/cuda/cuda_config.sh
index 0342985..0b9a53c 100755
--- a/third_party/gpus/cuda/cuda_config.sh
+++ b/third_party/gpus/cuda/cuda_config.sh
@@ -126,6 +126,8 @@ if test -e ${CUDNN_INSTALL_PATH}/libcudnn.so.6.5; then
   CUDNN_LIB_PATH=${CUDNN_INSTALL_PATH}
 elif test -e ${CUDNN_INSTALL_PATH}/lib64/libcudnn.so.6.5; then
   CUDNN_LIB_PATH=${CUDNN_INSTALL_PATH}/lib64
+elif test -e ${CUDNN_INSTALL_PATH}/lib/libcudnn.so.6.5; then
+  CUDNN_LIB_PATH=${CUDNN_INSTALL_PATH}/lib
 else
   CudnnError "cannot find libcudnn.so.6.5 under: ${CUDNN_INSTALL_PATH}"
 fi
